java-core/src/main/java/com/pcloud/sdk/internal/RealApiClient.java

```diff
@@ -172,10 +172,12 @@ public RemoteFolder adapt(Response response) throws IOException, ApiError {
 
     @Override
     public Call<RemoteFolder> listFolder(String path, boolean recursively) {
-
+        if (path == null) {
+            throw new IllegalArgumentException("Path argument cannot be null.");
+        }
         HttpUrl.Builder urlBuilder = apiHost.newBuilder()
                 .addPathSegment("listfolder")
-                .addEncodedQueryParameter("path", String.valueOf(path))
+                .addEncodedQueryParameter("path", path.isEmpty() ? "/" : path)
```

Comment:
Since all folders are loaded without trailing slash empty string should result in loading the root folder.

If this is something you want to have fixed on the server side, let me know.

Reply:
Let's say you want to `listFolder` `/Test/Sub Folder`. There is no trailing slash meaning `listFolder("/Test/Sub Folder/")` results in an error.

To keep this consistent, empty string should reference the root folder since `listFolder("/")` is the trailing slash.